File: src/main/java/fr/recia/glc/db/entities/personne/CategoriePersonne.java
/*
 * Copyright (C) 2023 GIP-RECIA, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package fr.recia.glc.db.entities.personne;

/**
 * Type énuméré des différents types de personnes.
 *
 * @author GIP RECIA - Gribonvald Julien
 * 10 juin 08
 */
public enum CategoriePersonne {
  /**
   * Un élève.
   */
  Eleve,
  /**
   * Un enseignant.
   */
  Enseignant,
  /**
   * Une personne non enseignante d'une collectivité locale.
   */
  Non_enseignant_collectivite_locale,
  /**
   * Une personne non enseignante faisant parti du personnel d'un établissement.
   */
  Non_enseignant_etablissement,
  /**
   * Une personne non enseignante d'un service académique.
   */
  Non_enseignant_service_academique,
  /**
   * Une personne étant uniquement une personne en relation avec un élève.
   */
  Personne_relation_eleve,
  /**
   * Une personne du personnel extérieur.
   */
  Personnel_exterieur,
  /**
   * Un responsable d'entreprise.
   */
  Responsable_Entreprise,
  /**
   * Un tuteur de stage.
   */
  Tuteur_stage
}

File: src/main/webapp/src/types/enums/CategoriePersonne.ts
import type { enumValues } from "@/types/enumValuesType";

export enum CategoriePersonne {
  Eleve = "Eleve",
  Enseignant = "Enseignant",
  Non_enseignant_collectivite_locale = "Non_enseignant_collectivite_locale",
  Non_enseignant_etablissement = "Non_enseignant_etablissement",
  Non_enseignant_service_academique = "Non_enseignant_service_academique",
  Personne_relation_eleve = "Personne_relation_eleve",
  Personnel_exterieur = "Personnel_exterieur",
  Responsable_Entreprise = "Responsable_Entreprise",
  Tuteur_stage = "Tuteur_stage",
}

const getCategoriePersonne = (categorie: string): enumValues => {
  switch (categorie) {
    case CategoriePersonne.Eleve.toString():
      return { i18n: "" };
    case CategoriePersonne.Enseignant.toString():
      return { i18n: "" };
    case CategoriePersonne.Non_enseignant_collectivite_locale.toString():
      return { i18n: "" };
    case CategoriePersonne.Non_enseignant_etablissement.toString():
      return { i18n: "" };
    case CategoriePersonne.Non_enseignant_service_academique.toString():
      return { i18n: "" };
    case CategoriePersonne.Personne_relation_eleve.toString():
      return { i18n: "" };
    case CategoriePersonne.Personnel_exterieur.toString():
      return { i18n: "" };
    case CategoriePersonne.Responsable_Entreprise.toString():
      return { i18n: "" };
    case CategoriePersonne.Tuteur_stage.toString():
      return { i18n: "" };
    default:
      throw new Error(
        `Non-existent categorie personne in switch: ${categorie}`
      );
  }
};

export { getCategoriePersonne };
